### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by jjsat**
- Fix crash in sys_tty_write() if the return buffer for the written length is NULL. Resident Evil 5 demo does this.

- ~~On Windows, pass the given file access modifiers like the Linux code. Dark Souls II creates a temporary file with write and append modifier, then tries to truncate the file. This needs GENERIC_WRITE.~~ Obsoleted by #3391.

- Endianess for the parameter buffer in setsockopt was not converted before passing the value on to the native function. Actually I don't know if converting the endianess is really necessary or if a true/false check with conversion to 1/0 would be enough or better. The value of these parameters can only be 0 or 1. This might break some games not because it is wrong but because they might not error out of the net code as early as now.

### CODE REVIEW

File: Utilities/File.cpp [hunk 1]
```diff
@@ -774,8 +774,9 @@ fs::file::file(const std::string& path, bs_t<open_mode> mode)
 #ifdef _WIN32
 	DWORD access = 0;
 	if (test(mode & fs::read)) access |= GENERIC_READ;
-	if (test(mode & fs::write)) access |= test(mode & fs::append) ? FILE_APPEND_DATA : GENERIC_WRITE;
-
+	if (test(mode & fs::write)) access |= GENERIC_WRITE;
+	if (test(mode & fs::append)) access |= FILE_APPEND_DATA;
```
Comment (Nekotekina) [inline Utilities/File.cpp]: Wrong, this won't work as on Linux.

Comment (jjsat) [reply Utilities/File.cpp]: The Linux code farther down does permit the same. On the PS3, the file is created like this and "truncated" (actually enlarged) to 10 MB. Emulator now would do the same.

Comment (Nekotekina) [reply Utilities/File.cpp]: You don't see what's happening: if write is NOT specified, append has no effect, but with your code it starts to "work".

Comment (Nekotekina) [reply Utilities/File.cpp]: Adding GENERIC_WRITE permission kills "atomic append" functionality: it doesn't write at the end of file anymore, but to the current Pointer instead. I'll try to investigate possible solution.

**Pull request merged**